docs/db/changelog/changesets/powerauth-webflow/1.5.x/20230510-spring-authorization-server-1.5.0.xml

```diff
@@ -120,8 +120,34 @@
+        <sql dbms="oracle">INSERT INTO oauth2_registered_client (id, client_id, client_id_issued_at, client_secret, client_secret_expires_at, client_name, client_authentication_methods, authorization_grant_types, redirect_uris, post_logout_redirect_uris, scopes, client_settings, token_settings) SELECT ROWNUM, client_id, CURRENT_TIMESTAMP(6), client_secret, null, client_id, 'client_secret_basic', authorized_grant_types, web_server_redirect_uri, null, scope, '{"@class":"java.util.Collections$UnmodifiableMap","settings.client.require-proof-key":false,"settings.client.require-authorization-consent":false}', '{"@class":"java.util.Collections$UnmodifiableMap","settings.token.reuse-refresh-tokens":true,"settings.token.id-token-signature-algorithm":["org.springframework.security.oauth2.jose.jws.SignatureAlgorithm","RS256"],"settings.token.access-token-time-to-live":["java.time.Duration",300],"settings.token.access-token-format":{"@class":"org.springframework.security.oauth2.server.authorization.settings.OAuth2TokenFormat","value":"reference"},"settings.token.refresh-token-time-to-live":["java.time.Duration",1296000],"settings.token.authorization-code-time-to-live":["java.time.Duration",300]}}' FROM oauth_client_details</sql>
+    </changeSet>
+
+    <!-- Migration to Spring authorization server: add constraint tpp_client_secret_fk in table tpp_app_detail -->
+    <changeSet id="6" logicalFilePath="powerauth-webflow/1.5.x/20230510-spring-authorization-server-1.5.0.xml" author="Roman Strobl">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <foreignKeyConstraintExists foreignKeyTableName="tpp_app_detail" foreignKeyName="tpp_client_secret_fk"/>
+            </not>
+        </preConditions>
+        <comment>Add constraint tpp_client_secret_fk in table tpp_app_detail</comment>
```

Comment:
The constraint should be added in ttp section of liquibase

Reply:
Ok, I can move it.